Align qemu_virt smp entry mailbox with arm-tf mailbox

etienne-lms opened the pull request:
Align qemu_virt smp entry mailbox with arm-tf mailbox
+
Prevent bootload issues when pager + secure data path are enabled.

jenswikl commented:
s/insure/ensure/g
with that fixed:
`Reviewed-by: Jens Wiklander <jens.wiklander@linaro.org>`

etienne-lms committed:
align bios smp entry mailbox to Arm trusted firmware mailbox

This allows OP-TEE to be booted in SMP from either bios_qemu_tz_arm
or arm trusted firmware.

Signed-off-by: Etienne Carriere <etienne.carriere@linaro.org>
Reviewed-by: Jens Wiklander <jens.wiklander@linaro.org>

etienne-lms commented:
commit comment fixed and tags applied.

jenswikl commented:
`Tested-by: Jens Wiklander <jens.wiklander@linaro.org>`

etienne-lms committed:
ensure bios never loads OP-TEE pagestore in SDP memory

Move OP-TEE page store load address to 4MB above start of reserved DRAM.
The offset 4MB is related to the first 1MB being reserved for early
boot stage (should be enough) and 2MB reserved to the SRAM emulation
in the memory layout. Offset is 4MB instead of 3MB to make it safer
and aligned on LPAE pgdir granule.

Prior this change, bios loaded the Op-tee pagestore near the end
of the secure DRAM which used as secure data path memory when SDP
support is enabled in OP-TEE.

Signed-off-by: Etienne Carriere <etienne.carriere@linaro.org>
Reviewed-by: Jens Wiklander <jens.wiklander@linaro.org>
Tested-by: Jens Wiklander <jens.wiklander@linaro.org>

etienne-lms commented:
tested-by tag applied.
ready to be merged together with OP-TEE/optee_os#2138.

jforissier commented:
> ready to be merged together with OP-TEE/optee_os#2138.

Is this incompatible with the current OP-TEE? (without 2138).
If so, https://github.com/OP-TEE/manifest/blob/master/default.xml#L16 will need updating, too, as soon as both PRs are merged.

etienne-lms commented:
I'm surprised the **default.xml** does not always point to the latest **bios_qemu_tz_arm**. There is no need to specify a sha1 for this component.

jforissier commented:
> I'm surprised the default.xml does not always point to the latest bios_qemu_tz_arm. There is no need to specify a sha1 for this component.

OK let's use master for this one too.

jforissier commented:
> OK let's use master for this one too.

Actually, not sure it's a good idea. If we do this we'll need to pin a revision or tag in [`make_stable.sh`](https://github.com/OP-TEE/manifest/blob/master/make_stable.sh). But we don't tag bios_qemu_tz_arm with OP-TEE version numbers currently (3.1.0, etc).
The simplest solution is to just update the SHA1.

jforissier commented:
https://github.com/OP-TEE/manifest/pull/105
